packages/experimental/src/smart-session/schema/index.ts

```diff
@@ -0,0 +1,158 @@
+import { z } from 'zod'
+import { Operation, ParamOperator } from '../utils/TypeUtils.js'
+
+// Define custom error messages
+export const ERROR_MESSAGES = {
+  UNSUPPORTED_NAMESPACE: 'Unsupported namespace',
+  NO_RESPONSE_RECEIVED: 'No response received from grantPermissions',
+  INVALID_REQUEST: 'Invalid request structure',
+  // GrantPermissionsRequest.chainId field
+  INVALID_CHAIN_ID_TYPE: 'Invalid chainId: type, expected string starting with "0x"',
+  INVALID_CHAIN_ID_FORMAT: 'Invalid chainId: must start with "0x"',
+  INVALID_CHAIN_ID_DATA: 'Invalid chainId format: Must be a hexadecimal string starting with "0x"',
+  // GrantPermissionsRequest.address field
+  INVALID_ADDRESS: 'Invalid address: must be a string starting with "0x"',
+  // GrantPermissionsRequest.expiry field
+  INVALID_EXPIRY: 'Invalid expiry: Expiry must be a future timestamp',
+
+  // GrantPermissionsRequest.signer field -  SignerSchema
+  INVALID_KEYS_SIGNER: 'A set of public keys is required for multiKey signers',
+  UNSUPPORTED_SIGNER_TYPE:
+    "Invalid signer.type: Invalid discriminator value. Expected 'key' | 'keys'",
+  UNSUPPORTED_KEY_TYPE: 'Unsupported key type: must be secp256r1 or secp256k1',
+  INVALID_PUBLIC_KEY_FORMAT: 'Invalid public key: must start with "0x"',
+  //PermissionSchema
+  INVALID_PERMISSIONS: 'Invalid permissions: must be a non-empty array',
+  INVALID_PERMISSIONS_TYPE: 'Invalid permissions: Expected array, received object',
+  INVALID_PERMISSIONS_TYPE_LITERALS: 'Invalid literal value, expected "contract-call"',
+  //PolicySchema
+  INVALID_POLICIES: 'Invalid policies: must be a non-empty array',
+  INVALID_POLICIES_TYPE: 'Invalid policies: Expected array, received object',
+
+  INVALID_GRANT_PERMISSIONS_RESPONSE: 'Invalid grantPermissions response'
+}
+
+// ChainId Schema
+const ChainIdSchema = z
+  .string({
+    invalid_type_error: ERROR_MESSAGES.INVALID_CHAIN_ID_TYPE
+  })
+  .refine(val => val.startsWith('0x'), {
+    message: ERROR_MESSAGES.INVALID_CHAIN_ID_FORMAT
+  })
+  .refine(val => /^0x[0-9a-fA-F]+$/u.test(val), {
+    message: ERROR_MESSAGES.INVALID_CHAIN_ID_DATA
+  })
+
+// Address Schema
```

Comment:
Comments are not giving any benefits, eighter we need to give more descriptive comments or remove them